### PR TITLE
Adds builds for databricks, trino and spark images

### PR DESCRIPTION
**Pull request by racheldaniel**
## What is this PR?
This is a:
- [ ] documentation update
- [ ] bug fix with no breaking changes
- [x] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation
Adds builds for databricks and spark images to github actions. This should not be merged until this PR has been approved and merged: https://github.com/dbt-labs/dbt-cloud-infra-terraform/pull/1122

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] Spark
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Review by isaac-taylor — APPROVED** [review]

**Comment by racheldaniel** [comment]
FYI merging this is blocked on this PR-- https://github.com/dbt-labs/dbt-cloud-infra-terraform/pull/1122 -- which has been merged but hasn't deployed due to a CI/CD issue on another PR in the Repo. Currently we're seeing the new `databricks`, `spark` and `trino` repos in `staging` and `dev`, but they need to be in `infra-root` as well before this gets merged.

File: .github/workflows/main.yml
Comment (racheldaniel) [reply .github/workflows/main.yml:102]: Ha at least in this repo 😄 Hopefully even simpler in the future!

**Pull request merged**
